swift-paperless/Views/TLSListView.swift

```diff
@@ -0,0 +1,231 @@
+                } footer: {
+                    switch certificateState {
+                    case .notloaded:
+                        Text(String(localized: .settings(.certificateNotLoaded)))
+                    case .wrongPassword:
+                        Text(String(localized: .settings(.certificateLoadError)))
+                    case .valid:
+                        HStack {
+                            Image("checkmark.circle.fill")
+                        }
+                    }
+                }
+
+                Section {
+                    Button(String(localized: .localizable(.save))) {
+                        guard let data = certificateData else {
+                            isCertificateValid = false
+                            certificateState = .notloaded
+                            return
+                        }
+                        saveToKeychain(certificateData: data, certificatePassword: certificatePassword, certificateName: certificateName)
+                        dismiss()
+                    }.disabled(!isCertificateValid)
+                }
+            }
+            .onChange(of: certificateData) {
+                validateInput()
+            }
+            .onChange(of: certificatePassword) {
+                validateInput()
+            }
+            .fileImporter(
+                isPresented: $isImporting,
+                allowedContentTypes: [.pkcs12],
+                allowsMultipleSelection: false
+            ) { result in
+                do {
+                    guard let selectedFile: URL = try result.get().first else { return }
+                    if selectedFile.startAccessingSecurityScopedResource() {
+                        certificateData = try Data(contentsOf: selectedFile)
+                        defer { selectedFile.stopAccessingSecurityScopedResource() }
+                        certificateName = selectedFile.lastPathComponent
+                    } else {
+                        Logger.shared.error("Error opening File from secure context")
+                    }
+                } catch {
+                    Logger.shared.error("erro while loding PfX \(error)")
+                }
+            }
+        }
+    }
+}
+
+struct TLSSingleView: View {
+    @State var identity: TLSIdentity
+    @State var cn: String?
+
+    var body: some View {
+        Form {
+            Section {
+                LabeledContent(LocalizedStringKey(localizable: .name), value: identity.name)
+                LabeledContent("CN", value: cn ?? "N/A")
+            }
+        }
+        .onChange(of: identity, initial: true) {
+            var optCertificate: SecCertificate?
+            SecIdentityCopyCertificate(identity.identity, &optCertificate)
+
+            if let certificate = optCertificate {
+                var cn: CFString?
+                SecCertificateCopyCommonName(certificate, &cn)
+
+                if cn != nil {
+                    self.cn = cn as String?
```

Comment:
Not super important, but I guess this could be a computed property of `TLSIdentity`.